packages/components/list-item/src/list-item.js

```diff
@@ -37,13 +41,24 @@ customElementDefineHelper(
 			return colorType;
 		}
 
+		slotChangeHandler(e) {
+			this.hasSlottedIcon = true;
+			const slottedNodes = e.currentTarget.assignedNodes();
+			validateSlottedNodes(this.tagName, slottedNodes, ['TS-APP-ICON']);
+			this.requestUpdate();
+		}
+
 		get iconLeftTemplate() {
 			const icon = this.icon || this.iconLeft;
-			if (icon) {
-				return html`
-					<ts-icon class="icon-left" icon="${icon}" size="large" type="${this.colorType}"></ts-icon>
-				`;
-			}
+			return icon
+				? html`
+						<ts-icon class="icon-left" icon="${icon}" size="large" type="${this.colorType}"></ts-icon>
+				  `
+				: html`
+						<span class="${this.hasSlottedIcon ? 'icon-left' : ''}">
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3AS3358)

Reply:
Also maybe you can use variables instead to make it easier to read and understand 👍